bee-tangle/src/tangle.rs

```diff
@@ -489,27 +489,49 @@ impl<B: StorageBackend> Tangle<B> {
     where
         Update: FnOnce(&mut MessageMetadata) -> R,
     {
-        let exists = self.pull_message(message_id, true).await;
+        let entry = self.vertices.entry(*message_id).await;
+        let was_vacant = matches!(entry, Entry::Vacant(_));
+        let mut vertex = entry.or_empty();
+
+        let mut perform_eviction = false;
+        let mut parents = None;
+
+        if was_vacant || vertex.message().is_none() {
+            if let Ok(Some((msg, metadata))) = self.storage_get(message_id) {
+                vertex.prevent_eviction();
```

Comment:
@thibault-martinez do you think this actually matters? there is no await point between `prevent_eviction` and `allow_eviction` and we are holding the guard to the partition via `vertex`.